code/modules/mob/living/carbon/human/species/unathi.dm

```diff
@@ -60,46 +60,43 @@
 
 /datum/species/unathi/on_species_gain(mob/living/carbon/human/H)
 	..()
-	var/datum/action/innate/tail_lash/lash = new()
-	lash.Grant(H)
+	var/datum/action/innate/ignite/fire = new()
+	fire.Grant(H)
 
 /datum/species/unathi/on_species_loss(mob/living/carbon/human/H)
 	..()
-	for(var/datum/action/innate/tail_lash/lash in H.actions)
-		lash.Remove(H)
+	for(var/datum/action/innate/ignite/fire in H.actions)
+		fire.Remove(H)
 
-/datum/action/innate/tail_lash
-	name = "Tail lash"
-	icon_icon = 'icons/effects/effects.dmi'
-	button_icon_state = "tail"
-	check_flags = AB_CHECK_LYING | AB_CHECK_CONSCIOUS | AB_CHECK_STUNNED
+/datum/action/innate/ignite
+	name = "Ignite"
+	desc = "Requires you to drink welding fuel beforehand."
+	icon_icon = 'icons/obj/cigarettes.dmi'
+	button_icon_state = "match_unathi"
+	check_flags = AB_CHECK_HANDS_BLOCKED | AB_CHECK_CONSCIOUS | AB_CHECK_STUNNED
 
-/datum/action/innate/tail_lash/Activate()
+
+/datum/action/innate/ignite/Activate()
 	var/mob/living/carbon/human/user = owner
-	if((user.restrained() && user.pulledby) || user.buckled)
-		to_chat(user, "<span class='warning'>You need freedom of movement to tail lash!</span>")
-		return
-	if(user.getStaminaLoss() >= 50)
-		to_chat(user, "<span class='warning'>Rest before tail lashing again!</span>")
-		return
-	for(var/mob/living/carbon/human/C in orange(1))
-		var/obj/item/organ/external/E = C.get_organ(pick("l_leg", "r_leg", "l_foot", "r_foot", "groin"))
-		if(E)
-			user.changeNext_move(CLICK_CD_MELEE)
-			user.visible_message("<span class='danger'>[user] smacks [C] in [E] with their tail! </span>", "<span class='danger'>You hit [C] in [E] with your tail!</span>")
-			user.adjustStaminaLoss(15)
-			C.apply_damage(5, BRUTE, E)
-			user.spin(20, 1)
-			playsound(user.loc, 'sound/weapons/slash.ogg', 50, 0)
-			add_attack_logs(user, C, "tail whipped")
-			if(user.restrained())
-				if(prob(50))
-					user.Weaken(10 SECONDS)
-					user.visible_message("<span class='danger'>[user] loses [user.p_their()] balance!</span>", "<span class='danger'>You lose your balance!</span>")
-					return
-			if(user.getStaminaLoss() >= 60) //Bit higher as you don't need to start, just would need to keep going with the tail lash.
-				to_chat(user, "<span class='warning'>You run out of momentum!</span>")
-				return
+	if(user.reagents && user.reagents.has_reagent("fuel", 3))
+		var/obj/item/match/unathi/fire = new(user.loc, src)
+		if((user.head && (user.head.flags_cover & HEADCOVERSMOUTH)) || (user.wear_mask && (user.wear_mask.flags_cover & MASKCOVERSMOUTH) && !user.wear_mask.up))
+			qdel(fire)
+			to_chat(user, "<span class='warning'>Your mouth is covered.</span>")
+		else
+			if(user.put_in_hands(fire))
+				to_chat(user, "<span class='notice'>You ignite a small flame in your mouth.</span>")
+				user.reagents.del_reagent("fuel")
```

Comment:
make this only remove the amount of fuel that the ability consumes so it doesn't purge all 330u of welding fuel from a welding fuel enjoyer or something